include/networkit/generators/RmatGenerator.hpp

```diff
@@ -39,14 +48,20 @@ class RmatGenerator final : public StaticGraphGenerator {
      * @param[in] d Probability for quadrant lower right
      * @param[in] weighted  result graph weighted?
      * @param[in] reduceNodes  number of random nodes to delete to achieve a given node count
+     * @param[in] discardSelfLoops ignore self loops
      */
     RmatGenerator(count scale, count edgeFactor, double a, double b, double c, double d,
-                  bool weighted = false, count reduceNodes = 0);
+                  bool weighted = false, count reduceNodes = 0, bool discardSelfLoops = true);
 
     /**
      * @return Graph to be generated according to parameters specified in constructor.
      */
     Graph generate() override;
+
+private:
+    void sample(std::mt19937_64 &urng);
+
+    std::pair<node, node> sampleEdge(std::mt19937_64 &urng, uint8_t bits);
```

Comment:
Check whether `urng` can be moved to a member variable. This is what is normally done in other algorithm classes. This would then also remove the need for passing it as a parameter.